pallets/statistics/src/lib.rs

```diff
@@ -276,28 +276,29 @@ impl<T: Config> Module<T> {
         let did = Self::ensure_asset_perms(origin, asset)?;
 
         // TODO: Use complexity instead of count to limit TransferConditions per asset.
-        // Check maximum TransferConditions per Asset limit.
-        ensure!(
-            transfer_conditions.len() < T::MaxTransferConditionsPerAsset::get() as usize,
-            Error::<T>::TransferConditionLimitReached
-        );
+        // converting from a btreeset to a bounded version
+        let bounded_transfer_conditions: BoundedBTreeSet<_, T::MaxTransferConditionsPerAsset> =
+            transfer_conditions
+                .clone()
```

Comment:
This clone shouldn't be needed, since we don't want to use the old `transfer_conditions` value anymore after converting it.  Only use the bounded version.  It would be better to replace the old variable `transfer_conditions` with the new bounded type, then it isn't possible to use the unbounded value later.

```suggestion
        let transfer_conditions: BoundedBTreeSet<_, T::MaxTransferConditionsPerAsset> =
            transfer_conditions
```